py/dml/c_backend.py

```diff
@@ -778,17 +778,22 @@ def generate_implement_method(device, ifacestruct, meth, indices):
         require_fully_typed(None, meth)
 
         # Calculate the expected method signature
-        ifacemethtype = ifacestruct.get_member_qualified(meth.name)
-        if not ifacemethtype:
+        member_type = ifacestruct.get_member_qualified(meth.name)
+        if not member_type:
             raise EMEMBER(meth.site, meth.parent.name, meth.name)
-        ifacemethtype = safe_realtype(ifacemethtype)
-        if isinstance(ifacemethtype, TPtr):
-            ifacemethtype = ifacemethtype.base
-        if not isinstance(ifacemethtype, TFunction):
```

Comment:
wait so the code was actually _intentionally written to support it?_ why the hell...?

Reply:
A guess:
1. some user sees "wow, I can express interface struct types as `void f(conf_object_t*);` so they look like normal C function declarations", and this worked out-of-the-box for `interface` objects (GCC doen't care, and not much typechecking was done by DMLC)
2. some user notices that such interfaces don't work with `implement` and reports a bug on DMLC
3. the above is a simple fix for that bug